src/scripts/main.js

```diff
@@ -1,3 +1,214 @@
+`;
+body.appendChild(form);
+
+// BUTTON
+
+const button = body.querySelector("[type='submit']");
+const createRow = (data) => {
+  const row = document.createElement('tr');
+
+  const convertToString = (num) => {
+    if (num === null || num === undefined || isNaN(+num)) {
+      return '';
+    }
+
+    return `$${(+num).toLocaleString('en-US')}`;
+  };
+
+  row.innerHTML = `
+    <td>${data.name}</td>
+    <td>${data.position}</td>
+    <td>${data.office}</td>
+    <td>${data.age}</td>
+    <td>${convertToString(data.salary)}</td>
+  `;
+
+  tbody.appendChild(row);
+  tbody = table.tBodies[0];
+  tbodyRows = [...tbody.rows];
+};
+// NOTIFICATION
+const pushNotification = (posRight, title, description, type) => {
+  const posTop = body.offsetHeight / 2 - body.getBoundingClientRect().top;
+  const notification = document.createElement('div');
+  const h2 = document.createElement('h2');
+  const p = document.createElement('p');
+
+  notification.classList.add('notification');
+  notification.classList.add(type);
+  notification.setAttribute('data-qa', 'notification');
+  notification.style.top = `${posTop}px`;
+  notification.style.right = `${posRight}px`;
+
+  h2.textContent = title;
+  p.textContent = description;
+  notification.append(h2, p);
+
+  body.appendChild(notification);
+  setTimeout(() => (notification.style.display = 'none'), 2000);
+};
+
+button.addEventListener('click', (e) => {
+  e.preventDefault();
+
+  const formData = new FormData(form);
+  const data = Object.fromEntries(formData.entries());
+
+  const isValide = () => {
+    const checkInput = Object.values(data).every((val) => val !== '');
+    const checkName = data.name.length >= 4;
+    const checkAge = data.age >= 18 && data.age <= 90;
+
+    if (checkInput && checkName && checkAge) {
+      return true;
+    }
+
+    return false;
+  };
+
+  if (isValide()) {
+    createRow(data);
+    pushNotification(10, 'Success', 'Data added successfully', 'success');
+    form.reset();
+  } else {
+    pushNotification(
+      10,
+      'Error',
+      'Please fill in all fields and check valid values',
+      'error',
+    ); // eslint-disable-line
+  }
+});
+
+// DBLCLICK
+
+// eslint-disable-next-line no-shadow
+tbody.addEventListener('dblclick', (event) => {
+  const cell = event.target.closest('td');
+
+  if (cell && cell.cellIndex !== undefined) {
+    const initialValue = cell.textContent.trim();
+    const input = document.createElement('input');
+
+    input.type = 'text';
+    // input.value = initialValue.replace(/[$,]/g, '');
+    input.classList.add('cell-input');
+    cell.innerHTML = '';
+    cell.appendChild(input);
+    input.focus();
+
+    const saveValue = () => {
+      const newValue = input.value.trim();
+
+      if (cell.cellIndex === 4) {
+        // Перевіряємо чи введено коректне числове значення
+        const parsedValue = parseFloat(newValue.replace(/[$,]/g, ''));
+
+        if (isNaN(parsedValue) || parsedValue < 0) {
```

Comment:
The salary validation logic should ensure that the input is a valid number and formatted correctly as a currency. Consider adding checks to handle edge cases, such as empty strings or non-numeric input.